mycroft/util/log.py

```diff
@@ -14,33 +14,116 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
-import json
+import inspect
 import logging
+import sys
+from traceback import format_exception
 
 from os.path import isfile
+
 from mycroft.util.json_helper import load_commented_json
 
-SYSTEM_CONFIG = '/etc/mycroft/mycroft.conf'
 
-__author__ = 'seanfitz'
+def getLogger(name="MYCROFT"):
+    """Depreciated. Use LOG instead"""
+    return logging.getLogger(name)
 
-log_level = "DEBUG"
 
-if isfile(SYSTEM_CONFIG):
-    config = load_commented_json(SYSTEM_CONFIG)
-    log_level = config.get("log_level", "DEBUG")
+class LOG:
+    """
+    Custom logger class that acts like logging.Logger
+    The logger name is automatically generated by the module of the caller
+
+    Usage:
+        LOG.debug('My message: %s', debug_str)
+        LOG('custom_name').debug('Another message')
+    """
 
-FORMAT = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-logging.basicConfig(format=FORMAT, level=logging.getLevelName(log_level))
-logger = logging.getLogger("MYCROFT")
+    _custom_name = None
 
+    @classmethod
+    def init(cls):
+        sys_config = '/etc/mycroft/mycroft.conf'
+        level_name = load_commented_json(sys_config)['log_level'] \
```

Comment:
I believe this will fail if a sys config exitst but lacks a `log_level` entry  should probably have `.get('log_level', 'info')` or similar.

Reply:
Here's my take on the subject:

Do we want to support there being a malformed system config? I'm pretty sure if someone deleted that there's a ton of other places in the code it would fail. Since the config gets updated at the same time as the code, I would think we should be able to assume it to always be there. Also, in the case that in the future a developer moves the log_level config value but forgets to update it here, rather than subtly introducing a big it will error out at the exact spot in the code that needs to be fixed.

Reply:
I'm not sure if a missing entry is considered malformed.

The `/etc/mycroft/mycroft.conf` is not used in any setups based on the github repo and the current template for the file does not include the log_level parameter so it would break logging on most installations.

Reply:
Yeah, you're right; I didn't realize it's only loading the system config. Will change.